toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -368,10 +370,10 @@ func convertImageToRaw(inputImageFile string, inputImageFormat string, rawImageF
 
 	err = shell.ExecuteLiveWithErr(1, "qemu-img", "convert", "-O", "raw", "--image-opts", sourceArg, rawImageFile)
 	if err != nil {
-		return fmt.Errorf("failed to convert image file to raw format:\n%w", err)
+		return "", fmt.Errorf("failed to convert image file to raw format:\n%w", err)
 	}
 
-	return nil
+	return imagecustomizerapi.ImageFormatType(detectedImageFormat), nil
```

Comment:
The strings used by qemu-img are different from the one used in `ImageFormatType`. `vpc` and `vhd` being the main difference. So, you can't cast the types.